app/src/app/[lng]/auth/signup/page.tsx

```diff
@@ -42,11 +43,20 @@ export default function Signup({
     handleSubmit,
     register,
     setError: setFormError,
+    setValue,
     formState: { errors, isSubmitting },
   } = useForm<Inputs>();
 
   const [error, setError] = useState("");
 
+  const searchParams = useSearchParams();
+  const params = searchParams.get("callbackUrl");
+  const inventory = params?.split("/").pop();
```

Comment:
This could potentially break if there are other arguments/ ? or # at the end of the URL. You can use the built-in `URL` class to parse an URL and extract the `pathname`, which you can then call split and pop on.

Details here: https://developer.mozilla.org/en-US/docs/Web/API/URL/pathname